### PR TITLE
Code for prompt monitoring of DoubleMu3_PFMET50 PFMHT60 path

### PR DESCRIPTION
**Pull request by gkaratha**
This code is for prompt monitoring of doublemu3_pfmet50 path. the MET and muon leg efficiencies are derived. the code compiles and passes the runTheMatrix test in CMSSW 9_2_1.

### CODE REVIEW

**Comment by mtosi** [comment]
@gkaratha please, rebase

**Comment by mtosi** [comment]
can you please summarize the list of trigger paths meant to be monitored by this PR ?
thanks

**Comment by mtosi** [comment]
tracked by https://github.com/cms-sw/cmssw/issues/19142

File: DQMOffline/Trigger/python/SoftOSMonitor_cff.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import FWCore.ParameterSet.Config as cms
+
+from DQMOffline.Trigger.SoftOSMonitor_cfi import hltSOSmonitoring
+
+# HLT_PFMETNoMu90_PFMHTNoMu90_IDTight
+DoubleMu3_PFMET50_DZ_PFMHT60_METmonitoring = hltSOSmonitoring.clone()
+DoubleMu3_PFMET50_DZ_PFMHT60_METmonitoring.FolderName = cms.string('HLT/SOS/MET/')
+DoubleMu3_PFMET50_DZ_PFMHT60_METmonitoring.numGenericTriggerEventPSet.hltPaths = cms.vstring("HLT_DoubleMu3_PFMET50_DZ_PFMHT60_v1")
```
Comment (mtosi) [inline DQMOffline/Trigger/python/SoftOSMonitor_cff.py]: change to v*, please !

**Comment by gkaratha** [comment]
Dear Mia,

Only our path DoubleMu3_PFMET50... Ok what "rebase" means? Thanks.

Best,
George

File: DQMOffline/Trigger/python/SoftOSMonitor_cff.py (hunk 1 above)
Comment (mtosi) [inline DQMOffline/Trigger/python/SoftOSMonitor_cff.py]: in addition, why I do not find this path in silvio's google doc ?

Comment (mtosi) [inline DQMOffline/Trigger/python/SoftOSMonitor_cff.py]: I find 

HLT_DoubleMu3_DZ_PFMET50_PFMHT60_v*
HLT_DoubleMu3_DZ_PFMET70_PFMHT70_v*
HLT_DoubleMu3_DZ_PFMET90_PFMHT90_v*


!!!!!!!!!!!!!!!!!!!!

**Comment by mtosi** [comment]
I'm confused :(
did not we say to try to go w/ a slightly different approach ?

**Comment by gkaratha** [comment]
Dear Mia,

Ok sorry by DoubleMu3_PFMET50 I meant DoubleMu3_DZ_PFMET50_PFMHT60. I have modified the TopMonitoring and passes the tests. As I sent you it has some conflicts... Should I make the modified topMonitoring a PR instead?Thanks.

Best,
George

**Comment by gkaratha** [comment]
It is already pushed.. it will take 2 min

**Comment by mtosi** [comment]
git rebase -i CMSSW_9_3_X

**Comment by mtosi** [comment]
please
1. fix the name of trigger path !
2. I see this PR is adding DQMOffline/Trigger/plugins/SOSMonitor.cc
?!?!?!?!?!?
=> I'm really really really confused (sorry)

**Comment by gkaratha** [comment]
If i rebase i get this:
fatal: Needed a single revision
invalid upstream CMSSW_9_3_X

Yes because this is what i pushed... Now i am confused too... yesterday i thought i had to take the old PR and make it in cmssw 92x 93x... We exchanged email for this.. Was that wrong?

Best,
george

**Comment by mtosi** [comment]
if you are in the branch for 92x
git rebase -i CMSSW_9_2_X
if you are in the branch for 93x (master)
git rebase -i CMSSW_9_3_X

no ?
  mia

**Comment by gkaratha** [comment]
Dear Mia, many thanks for your help!
Is that what i am supposed to see?

**Comment by mtosi** [comment]
I think you need to rebase ....

**Comment by gkaratha** [comment]
I just did...
If I do it now i get this:
Successfully rebased and updated refs/heads/DQM_for_DoubleMu3_PFMET50.

**Comment by mtosi** [comment]
did you do w.r.t. 93x or 92 x ?
there are more than 20 files in this PR !!!!!

**Comment by gkaratha** [comment]
Yes because this is what I PRed.. The TopMonitoring is pushed but not PRed, that was the question I have been asking... The other paths are backups so they will be used only if the first fails...Should be monitored too? Or we can just change them when the time comes? 92X...

**Comment by slava77** [comment]
-1

irrelevant commits in the history. Either a rebase or other changes is needed
http://cms-sw.github.io/tutorial-resolve-conflicts.html may help

**Comment by fwyzard** [comment]
@gkaratha , you need to update the local copy of the git repository in your working area before rebasing, in order to pick up the latest version of the upstream (official) CMSSW branches. 

For example you could do

```
cmsenv
git fetch official-cmssw
git rebase official-cmssw/CMSSW_9_2_X
```

Ciao,
.Andrea

**Comment by gkaratha** [comment]
Dear Andrea,

Many thanks for your advice! I appreciate it. I manage (with suggestions from Mia) to make another PR which is here: https://github.com/cms-sw/cmssw/pull/19627#issuecomment-313870053 (i wrote in the JIRA as well). The new PR is a modification of another monitoring code and the reason of this change is not a bug in my previous code but rather the fact that we want to reduce the number of monitoring codes.  I am in the middle of closing all the previous PRs to avoid the confusion. Thanks again.

Many thanks and best regards,
George

**Pull request closed without merging**